File: Server/fe/src/services/pdf.js
import { NextResponse } from 'next/server'
import { readFileSync } from 'fs'
import path from 'path'

export async function GET(request, { params }) {
  try {
    const { filename } = params
    
    console.log('PDF API called with filename:', filename)
    console.log('Current working directory:', process.cwd())
    
    // Validate filename
    const allowedFiles = ['taiLieuBaoTri.pdf', 'linhKien.pdf']
    
    if (!allowedFiles.includes(filename)) {
      console.error('Filename not allowed:', filename)
      return NextResponse.json({ error: 'File not found' }, { status: 404 })
    }
    
    // Đường dẫn tới file PDF - sử dụng filename trực tiếp (đã copy file không dấu)
    const filePath = path.join(process.cwd(), 'pdf', filename)
    console.log('Attempting to read file from:', filePath)
    
    try {
      const fileBuffer = readFileSync(filePath)
      console.log('File read successfully, size:', fileBuffer.length)
      
      // Map display names
      const displayNames = {
        'taiLieuBaoTri.pdf': 'taiLieuBaoTri.pdf',
        'linhKien.pdf': 'linhKien.pdf'
      }
      
      const displayName = displayNames[filename] || filename
      
      return new NextResponse(fileBuffer, {
        headers: {
          'Content-Type': 'application/pdf',
          'Content-Disposition': `inline; filename="${encodeURIComponent(displayName)}"`,
          'Cache-Control': 'public, max-age=31536000',
        },
      })
    } catch (fileError) {
      console.error('Error reading PDF file:', fileError)
      console.error('File path:', filePath)
      return NextResponse.json({ 
        error: 'File not accessible',
        details: fileError.message,
        path: filePath 
      }, { status: 404 })
    }
    
  } catch (error) {
    console.error('Error serving PDF:', error)
    return NextResponse.json(
      { error: 'Failed to serve PDF file', details: error.message },
      { status: 500 }
    )
  }
}

File: Server/be/app/routers/pdf.py
from fastapi import APIRouter, HTTPException
from fastapi.responses import FileResponse
import os
from pathlib import Path

router = APIRouter()

# Đường dẫn tới thư mục PDF (tương đối với file này)
PDF_DIR = Path(__file__).parent.parent.parent / "pdf"

# Danh sách file PDF được phép
ALLOWED_FILES = ["taiLieuBaoTri.pdf", "linhKien.pdf"]

@router.get("/api/pdf/{filename}")
async def get_pdf(filename: str):
    """
    Serve PDF files từ thư mục pdf/
    
    Args:
        filename: Tên file PDF (taiLieuBaoTri.pdf hoặc linhKien.pdf)
    
    Returns:
        FileResponse: PDF file
    """
    # Validate filename
    if filename not in ALLOWED_FILES:
        raise HTTPException(
            status_code=404,
            detail=f"File không được phép: {filename}. Chỉ cho phép: {', '.join(ALLOWED_FILES)}"
        )
    
    # Tạo đường dẫn đầy đủ
    file_path = PDF_DIR / filename
    
    # Kiểm tra file có tồn tại không
    if not file_path.exists():
        raise HTTPException(
            status_code=404,
            detail=f"File không tìm thấy: {filename}"
        )
    
    # Trả về file PDF với Content-Type phù hợp
    return FileResponse(
        path=file_path,
        media_type="application/pdf",
        filename=filename,
        headers={
            "Content-Disposition": f"inline; filename={filename}"
        }
    )

